guides/v2.3/graphql/mutations/add-downloadable-products.md

```diff
@@ -0,0 +1,255 @@
+              {
+                "title": "Trailer #3",
+                "sample_url": "https://<M2_INSTANCE>/downloadable/download/sample/sample_id/18/"
+              }
+            ]
+          }
+        ]
+      }
+    }
+  }
+}
+```
+
+{: .bs-callout-info }
+
+### Add a downloadable product to a cart with disabled `Links can be purchased separately`
+
+The following example displays how to add a downloadable product with the `sku` value of `240-LV07` from the Luma sample data into shopping cart. This product has disabled `Links can be purchased separately` option, so all downloadble links will be added into cart automatically. 
+
+**Request**
+
+```text
+mutation {
+  addDownloadableProductsToCart(
+    input: {
+      cart_id: "gMV2BFQuNGiQmTnepQlMGko7Xc4P3X1w"
+      cart_items: {
+        data: {
+          sku: "240-LV07",
+          quantity: 1
+        }
+      }
+    }
+  ) {
+    cart {
+      items {
+        product {
+          sku
+        }
+        quantity
+        ... on DownloadableCartItem {
+          links {
+            title
+            price
+          }
+          samples {
+            title
+            sample_url
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+**Response**
+
+```text
+{
+  "data": {
+    "addDownloadableProductsToCart": {
+      "cart": {
+        "items": [
+          {
+            "product": {
+              "sku": "240-LV07"
+            },
+            "quantity": 2,
+            "links": [
+              {
+                "title": "Solo Power Circuit",
+                "price": 14
+              }
+            ],
+            "samples": [
+              {
+                "title": "Trailer #1",
+                "sample_url": "https://<M2_INSTANCE>/downloadable/download/sample/sample_id/10/"
+              },
+              {
+                "title": "Trailer #2",
+                "sample_url": "https://<M2_INSTANCE>/downloadable/download/sample/sample_id/11/"
+              },
+              {
+                "title": "Trailer #3",
+                "sample_url": "https://<M2_INSTANCE>/downloadable/download/sample/sample_id/12/"
+              }
+            ]
+          }
+        ]
+      }
+    }
+  }
+}
+```
+
+## Input attributes
+
+The top-level `AddDownloadableProductsToCartInput` object is listed first. All child objects are listed in alphabetical order.
+
+### AddDownloadableProductsToCartInput object {#AddDownloadableProductsToCartInput}
+
+The `AddDownloadableProductsToCartInput` object must contain the following attributes:
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`cart_id` | String! | The unique ID that identifies the customer's cart
+`cart_items` | [DownloadableProductCartItemInput!](#DownloadableProductCartItemInput) | Contains the cart item IDs and quantity of each item
+
+### DownloadableProductCartItemInput object {#DownloadableProductCartItemInput}
+
+The `DownloadableProductCartItemInput` object must contain the following attributes:
```

Comment:
We say that the object must contain the following attributes, however, `downloadable_product_links` and `customizable_options` are optional attributes. So "must" is not the 100% correct assumption here.

Reply:
Really a good point.

**fixed**